packages/fxa-settings/src/lib/hooks.tsx

```diff
@@ -50,3 +62,22 @@ export function useChangeFocusEffect() {
 
   return elToFocus;
 }
+
+// TODO - Iron out how we want to handle mutation errors (FXA-2450)
+// We must have an `onError` option in mutations to allow tests to pass
+// but providing one prevents an error from actually throwing so we need
+// to manually report it to Sentry.
+// See https://github.com/apollographql/react-apollo/issues/2614
+export function useHandledMutation(
+  mutation: DocumentNode,
+  options: MutationHookOptions<any, Record<string, any>> | undefined = {}
+) {
+  return useMutation(
+    mutation,
+    Object.assign(options, {
+      onError: (error: ApolloError) => {
+        sentryMetrics.captureException(error);
+      },
```

Comment:
We want to be able to handle the error at the call location as well in order to handle mutation specific errors. If there's an `onError` in `options` we should wrap it - probably in a try/catch and let the passed in handler either handle it or rethrow to here. Some errors don't belong in sentry so we shouldn't report unless it's unexpected.

Reply:
Makes sense.